nautobot/extras/utils.py

```diff
@@ -880,3 +882,22 @@ def bulk_delete_with_bulk_change_logging(qs, batch_size=1000):
         finally:
             change_context.defer_object_changes = False
             change_context.reset_deferred_object_changes()
+
+
+def format_query_param_dict(param_dict, view_name, non_filter_params):
+    """
+    Called before saving query filter parameters to a SavedView's config. This function will format
+    single value query parameters to be saved as a single values instead of lists of singles values.
+
+    Args:
+        param_dict (dict): key-value pairs of query parameters.
+        view_name (str): The name of the view that the saved view is associated with. "dcim:location_list" for example.
+        non_filter_params (list): List of non-query parameters that should not be formatted.
+    """
+    model = get_model_for_view_name(view_name)
+    filterset = get_filterset_for_model(model)
+
+    for filter_field, value in param_dict.items():
+        if filter_field not in non_filter_params and is_single_choice_field(filterset(), filter_field):
```

Comment:
From the test failure in CI, looks like probably need a try/except here for filter_field names that aren't actually in the filterset